### DISCUSSION TITLE
Filter Presence Intelligence People

### DISCUSSION THREAD

**Discussion by ltraveler** [idena-network/idena-docs#119] [IIPs]
To make IDENA more Sybil resistant. We could create one more part in the validation ceremony to prove your physical presence.

**The scenario would be:**
- The same as we have the keywords for flips, we could prepare the list of gestures that user would have to make before the validation ceremony in front of the camera.
- The video stream gonna be saved in ipfs and limited by 5 seconds for example.
- After passing the live physical presence test the user gonna continue with the regular IDENA ceremony as we currently have
- The reporting part of the ceremony would have 1 more additional part which is to report videos that are not repeating textually described gesture, the same as we are reporting flips if they do not include all keywords.
- In case if identity won't reach the physical presence consensus the identity gonna fail the validation.
- In case if identity gonna reach a weak consensus the validation/mining rewards for that identity could be penalized.
- People who doesn't want to show their faces could pass the live physical presence part using a mask to hide the real face, the most important part is to repeat the gesture.
- All saved video stream of the whole network gonna be eliminated after reaching the consensus.

Example of gestures:
- Show specific number of fingers
- V gesture
- Open your mouth and close one of the eyes
- Some gestures from finger-language
- Hi gesture
- Namaste gesture
- OK gesture
- Thumbs up gesture
- Thumbs down gesture

The gestures must be repeated close to the face. The face could be covered by mask.

Those are the basic points of the Live Physical Presence Validation Part.
And it is a good start to share your thoughts about it.

**Comment by Zen-44** [comment]
Could you elaborate more on how this will make Idena more sybil resistant? Doesn't solving the flips already prove your physical presence at a device?

**Reply by ltraveler** [reply]
1. It wouldn't be possible to make multiple validations anymore. We will come closer to the basic principle of the network **1 vote - 1 identity**. No multiple validations - no multiple nodes with vote rights.
2. For farm owners it will become way harder to break that system, it will bring more financial charges and alienate many current workers because of the new requirements.
3. It will bring more people to the network because the validation ceremony will be pretty fun, not just solving flip images. Everyone and especially newbies would like to see all of that recorded videos knowing that the IDENA network is live. It will bring more trust to our project. As a consequence of the network organic growth based on inflow of new independent users, new people would like to run their independent nodes, publish ads. Also, new companies and investors will take notice on IDENA knowing that behind each of the identity there is a real person.

**Reply by Zen-44** [reply]
I do not think this will make the ceremony funnier overall. I think having to submit additional video verification will be seen as a burden. In my opinion, the implementation would be complicated both in terms of code development, as well as community adoption. 

I think the problem of one person solving flips for multiple identities would rather be solved by modifying the short session time limit if it was a significant issue that needed addressing.

**Comment by ltraveler** [comment]
> I do not think this will make the ceremony funnier overall.

It is not so important of what do you think or what I think if there is no constructive suggestion behind it. Every man to his own taste. The only important thing to move that proposal forward would be the community opinion at the origin.

> the implementation would be complicated both in terms of code development

Sounds like a guess work, Neither you nor I don't know how complicated would be the development and implementation process. Also, first thing first. Pilot Oracle with brief explanation of the Idea brings us the feedback and if the feedback gonna be positive then It could be represented as weighty argument to turn developer's mind to that proposal.

> as well as community adoption

Again, for now it is just statements not susceptible of proof. Only Oracle results could give us a real feedback from the community. Now, we are just 2 people and our guessing doesn't have much sense.

> rather be solved by modifying the short session time limit

I don't agree with that point because the spread in average values of people cognitive capability is too high and depends on many factors like age, chronic conditions, environment and so on. 
In contrast the live presence test would take specific timeframe to be focused only on proving your physical presence.  It will bring more trust between network members and also it gives more trust to sceptic people and companies who aware of farm existence and multi accounting possibility.

## In summary:
### 1st stage
Creating a Pilot Oracle with brief explanation of the idea with two simple options `Yes, I would like to pass FPIP` and `No, I wouldn't like to pass FPIP`.
I already start to draft the proposal into article on Medium to add the link in the Oracle description.

### 2nd stage
In case of positive feedback. The results of the Oracle could be represented as potential call from the community to add this feature into validation ceremony.

### 3rd stage
In case of the positive feedback from the developers core team. The whole integration process could be divided into small substages to make the implementation process as smooth as possible. With a sufficient test stage where people could try FPIP and adopt themselves into new live physical presence stage before the real implementation.

**Reply by Zen-44** [reply]
> It is not so important of what do you think or what I think if there is no constructive suggestion behind it. 

Than why mention that in the first place?

> Neither you nor I don't know how complicated would be the development and implementation process

I think most people who regularly follow commits of an open-source project and have a grasp of the way the code works can estimate the difficulty of a feature to be added.

> for now it is just statements not susceptible of proof

No one said this is proof of anything. It is only natural for opinions to be shared.


You should not take these as an attack, but rather as pure opinions.

**Reply by ltraveler** [reply]
> Than` why mention that in the first place?

The purpose of the topic is get constructive thoughts, ideas of how this draft could be evolve into the final draft proposal. I don't think so, that proving the point of view would it be fun or not could bring any practical sense. Some people would take it as fun and some not. The point is that potentially it will make the whole process more interactive and alive.  What I am expecting is a real constructive suggestions of what could be improved to make this idea more reliable and easily-deployable.

> I think most people who regularly follow commits of an open-source project and have a grasp of the way the code works can estimate the difficulty of a feature to be added.

In any project we have two sides: consumers in our case it is the community and developers who are making the product. Adding a new feature it is always complications. Before answering on the question "Is the game worth the candle?" You would have to answer on some questions:

1. What your consumers think about it?
2. How that feature improve the project?
3. What would be the potential benefits?
4. Would it increase the market capitalization?
5. Would it increase the investment attractive?
6. How much more hardware resources the network validators would consume?
7. How smoothly the whole implementation process would go?
etc.

Any difficulties could be overcome the question is how much staff-hours would it cost?  

So, in one side is potential benefits, wants and wishes and in the other is complications. When you have more or less the whole picture you could make a team meeting and make a constructive discussion to take a final decision.

From my side as a community member I would like to prepare the proposal and make a pilot Oracle to ask opinion of the community and represent this opinion to developers as call from the community but of course the final decision would be in the hands of the core team.

I have create this topic to know opinion of others of what could be improved here or changed to make the whole live presence verification proposal more solid and polished before to run the Pilot Oracle to get the feedback from the community.

> You should not take these as an attack, but rather as pure opinions.

I am ok to hear a healthy, well-founded criticism but not just `the implementation would be complicated` or `I do not think this will make the ceremony funnier overall`.

**Reply by Zen-44** [reply]
>I am ok to hear a healthy, well-founded criticism but not just the implementation would be complicated or I do not think this will make the ceremony funnier overall.

I am not sure how sharing my thoughts on these two things (ceremony being funnier; implementation process: code & community adaptation) is not healthy. I am sorry to hear this was unwelcome.

**Reply by ltraveler** [reply]
I am sorry, I didn't mean to make my reply uncourteous. I think it would be better if I will give an example of constructive criticism of complications in implementation process that came to my mind:

Make a video stream and save it will take more resources from mobile devices, also the standard of the saved video must be compatible with all type of devices, each of the member would need to have a camera in each of the ceremony, that brings more complications, for example some people do not have cameras on their laptops even some people using mobile devices with broken cameras.

As you can see there are some facts and we could start a constructive discussion based on them.
The primary objective is not to make me feel to give up but to form the foundation for the live presence proposal to represent it for the community and get the feedback in the final result of the Pilot Oracle.

**Comment by mnz2000** [comment]
It would be way too easy to generate a video with the required gestures programmatically. One could just make short videos of all individual gestures, then glue them together as required. With proper start and finish constraints (like hands out of the picture), this could be done even without any on-the-fly editing between gestures. If needed, the transitions between gestures could probably be generated automatically rather easily.

Obviously, this would require a modified client, but that's not any problem for serious farmers.
